nyxtools/nyx_raster_flyer.py

```diff
@@ -0,0 +1,167 @@
+        self.zebra.pc.arm_signal.put(1)
+        zebra_status.wait()
+        print(f"zebra arm status success: {zebra_status.success}")
+        self.vector.move()
+        return NullStatus()  # why not return the status object?
+
+    def update_parameters(self, *args, **kwargs):
+        logger.debug(f"starting updating parameters with {kwargs}")
+        self.configure_vector(**kwargs)
+        row_index = kwargs.get("row_index", 0)
+        kwargs["num_images"] = float(kwargs["num_images"])
+        numImages = kwargs["num_images"]
+
+        if row_index == 0:
+            print("row 0: fully configuring zebra")
+            self.zebra.pc.pulse.max.put(numImages)
+            self.configure_zebra(**kwargs)
+        else:
+            print(f"row {row_index}: only setting pulse max")
+            self.zebra.pc.pulse.max.put(numImages)
+        print("finished updating parameters")
+
+    def configure_detector(self, *args, **kwargs):
+        file_prefix = kwargs["file_prefix"]
+        data_directory_name = kwargs["data_directory_name"]
+        self.detector.file.external_name.put(file_prefix, wait=True)
+        self.detector.file.write_path_template = data_directory_name
+
+    def setup_zebra_vector_scan(
+        self,
+        angle_start,
+        gate_width,
+        scan_width,
+        pulse_width,
+        pulse_step,
+        exposure_period_per_image,
+        num_images,
+        is_still=False,
+    ):
+        print("setuip zebra vector scan")
+        self.zebra.pc.encoder.put(0, wait=True)  # 0 = omega for nyx
+        self.zebra.pc.direction.put(0, wait=True)  # 0 = positive trigger
+        self.zebra.pc.gate.start.put(angle_start, wait=True)
+        if is_still is False:
+            print(f"before: gate width: {gate_width} gate step: {scan_width}")
+            self.zebra.pc.gate.width.put(gate_width, wait=True)
+            self.zebra.pc.gate.step.put(scan_width, wait=True)
+        self.zebra.pc.gate.num_gates.put(1, wait=True)
+        # self.zebra.pc.pulse.start.put(0, wait=True)
+        self.zebra.pc.pulse.start.put(55, wait=True)
+        print(f"before: pulse width: {pulse_width}")
+        self.zebra.pc.pulse.width.put(pulse_width, wait=True)
+        self.zebra.pc.pulse.step.put(pulse_step, wait=True)
+        print(f"before: pulse delay: {exposure_period_per_image / 2 * 1000}")
+        self.zebra.pc.pulse.delay.put((exposure_period_per_image / 2 * 1000), wait=True)
+        print(
+            f"after: gate width: {self.zebra.pc.gate.width.get()} gate step: {self.zebra.pc.gate.step.get()}"
+            f"after: pulse width: {self.zebra.pc.pulse.width.get()} pulse delay: {self.zebra.pc.pulse.delay.get()}"
+        )
+        self.zebra.pc.pulse.max.put(num_images, wait=True)
+        # self.vector.hold.put(0)  # necessary to prevent problems upon
+        # exposure time change  elf.detector.cam.acquire.put(1)
+
+    def detector_arm(self, **kwargs):
```

Comment:
Taking `kwargs` this way is technically correct, but bad practice.  There are two major problems with this.  

First because there are (a bunch) of required keys it is impossible to tell that from looking at the signature (and the lack of docstring) so this reduces both the readability and the ability for static analysis to help out.

The second issue is that if you miss-spell a required key and provide one correct and one incorrect version the incorrect spelling will be silently accepted and ignored which can lead to very frustrating bugs.

Reply:
If these are a common set of inputs, then it may be worth creating a "configuration_class" (e.g. using `Dataclass` to pass around), but if not then you just just own the huge signature and write it out.  

You can still call it as `foo(**kwargs)` at the call site if needed.

Reply:
thanks for that suggestion, a `Dataclass` would be a good solution and better than having the huge signature, and could be set up with sensible default values.